File: app/src/main/java/com/example/cobe/cryptonews/ui/articles/ArticleViewHolder.java
package com.example.cobe.cryptonews.ui.articles;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.cobe.cryptonews.R;
import com.example.cobe.cryptonews.listeners.OnArticleClickListener;
import com.example.cobe.cryptonews.model.Article;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by cobe on 30/03/2018.
 */

public class ArticleViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    @BindView(R.id.ivArticleImage)
    ImageView articleImage;

    @BindView(R.id.tvArticleTitle)
    TextView articleTitle;

    @BindView(R.id.tvArticleDescription)
    TextView articleDescription;

    @BindView(R.id.tvArticleCreator)
    TextView articleCreator;

    @BindView(R.id.tvArticlePublishedAt)
    TextView articlePublishedAt;

    private OnArticleClickListener onArticleClickListener;
    private String url;

    public ArticleViewHolder(View itemView, OnArticleClickListener onArticleClickListener) {
        super(itemView);

        this.onArticleClickListener = onArticleClickListener;
        ButterKnife.bind(this, itemView);
    }

    public void setArticle(Article article) {
        url = article.getUrl();

        Glide.with(articleImage.getContext()).load(article.getUrlToImage()).into(articleImage);
        articleTitle.setText(article.getTitle());
        articleCreator.setText(article.getAuthor());
        articleDescription.setText(article.getDescription());

        String[] publishedDate = article.getPublishedAt().split("T");
        articlePublishedAt.setText(publishedDate[0]);

        itemView.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (onArticleClickListener != null) {
            onArticleClickListener.onArticleClick(url);
        }
    }
}

File: app/src/main/java/com/example/cobe/cryptonews/ui/articles/ArticleViewHolder.kt
package com.example.cobe.cryptonews.ui.articles

import android.support.v7.widget.RecyclerView
import android.view.View
import com.bumptech.glide.Glide
import com.example.cobe.cryptonews.listeners.OnArticleClickListener
import com.example.cobe.cryptonews.model.Article
import kotlinx.android.synthetic.main.article_item.view.*

/**
 * Created by cobe on 13/04/2018.
 */
class ArticleViewHolder(itemView: View, private var onArticleClickListener: OnArticleClickListener) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

    private lateinit var url: String

    fun setArticle(article: Article) {
        url = article.url

        Glide.with(itemView.articleImage.context).load(article.urlToImage).into(itemView.articleImage)
        itemView.articleTitle.text = article.title
        itemView.articleCreator.text = article.author
        itemView.articleDescription.text = article.description

        val publishedDate = article.publishedAt.split("T")
        itemView.articlePublishedAt.text = publishedDate[0]

        itemView.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        onArticleClickListener.onArticleClick(url)
    }
}